tests/components/dio_chacon/test_cover.py

```diff
@@ -0,0 +1,204 @@
+        {ATTR_ENTITY_ID: entity.entity_id},
+        blocking=True,
+    )
+    await hass.async_block_till_done()
+    state = hass.states.get("cover.shutter_mock_1")
+    assert state.state == STATE_CLOSING
+
+    await hass.services.async_call(
+        COVER_DOMAIN,
+        SERVICE_STOP_COVER,
+        {ATTR_ENTITY_ID: entity.entity_id},
+        blocking=True,
+    )
+    await hass.async_block_till_done()
+    state = hass.states.get("cover.shutter_mock_1")
+    assert state.state == STATE_OPEN
+
+    await hass.services.async_call(
+        COVER_DOMAIN,
+        SERVICE_OPEN_COVER,
+        {ATTR_ENTITY_ID: entity.entity_id},
+        blocking=True,
+    )
+    await hass.async_block_till_done()
+    state = hass.states.get("cover.shutter_mock_1")
+    assert state.state == STATE_OPENING
+
+    # mock_dio_chacon_client.move_shutter_direction.return_value = {}
+    await hass.services.async_call(
+        COVER_DOMAIN,
+        SERVICE_SET_COVER_POSITION,
+        {ATTR_POSITION: 25, ATTR_ENTITY_ID: entity.entity_id},
+        blocking=True,
+    )
+    await hass.async_block_till_done()
+    state = hass.states.get("cover.shutter_mock_1")
+    assert state.state == STATE_OPENING
+
+    # Server side callback tests
+    # We find the callback method on the mock client
+    callback_device_state_function: Callable = (
+        mock_dio_chacon_client.set_callback_device_state_by_device.call_args[0][1]
+    )
+    # And calls it to effectively launch the callback as the server would do
+    callback_device_state_function(
+        {
+            "id": "L4HActuator_idmock1",
+            "connected": True,
+            "openlevel": 79,
+            "movement": "stop",
+        }
+    )
+    await hass.async_block_till_done()
+    state = hass.states.get("cover.shutter_mock_1")
+    assert state
+    assert state.attributes.get(ATTR_CURRENT_POSITION) == 79
+    assert state.state == STATE_OPEN
+
+    callback_device_state_function(
+        {
+            "id": "L4HActuator_idmock1",
+            "connected": True,
+            "openlevel": 90,
+            "movement": "up",
+        }
+    )
+    await hass.async_block_till_done()
+    state = hass.states.get("cover.shutter_mock_1")
+    assert state
+    assert state.attributes.get(ATTR_CURRENT_POSITION) == 90
+    assert state.state == STATE_OPENING
+
+    callback_device_state_function(
+        {
+            "id": "L4HActuator_idmock1",
+            "connected": True,
+            "openlevel": 60,
+            "movement": "down",
+        }
+    )
+    await hass.async_block_till_done()
+    state = hass.states.get("cover.shutter_mock_1")
+    assert state
+    assert state.attributes.get(ATTR_CURRENT_POSITION) == 60
+    assert state.state == STATE_CLOSING
+
+    # Tests coverage for unload actions.
+    # mock_dio_chacon_client.disconnect.return_value = {}
+    await hass.config_entries.async_unload(mock_config_entry.entry_id)
+
+    # pytest.fail("Fails to display logs of tests")
```

Comment:
Let's put this in a separate test. Maybe wrap the `callback_device_state_function` in a function so you don't have to repeat the `id` and `connected` every call. In `knocki` I have something like that.

Reply:
I find it more readable to have explicit parameters in the various calls.